accounts/abi/bind/backends/simulated.go

```diff
@@ -109,12 +112,37 @@ func (b *SimulatedBackend) Commit() {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
-	if _, err := b.blockchain.InsertChain([]*types.Block{b.pendingBlock}); err != nil {
+	// Get the last block
+	parentBlock := b.blockchain.GetBlockByHash(b.pendingBlock.ParentHash())
+	var remainingTx types.Transactions
+
+	blocks, _ := core.GenerateChain(b.config, parentBlock, ethash.NewFaker(), b.database, 1,
+		func(number int, block *core.BlockGen) {
+
+			for _, tx := range b.pendingBlock.Transactions() {
+				if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice()) <= 0 {
+					block.AddTxWithChain(b.blockchain, tx)
+				} else {
+					remainingTx = append(remainingTx, tx)
+				}
+			}
+
+			for _, tx := range b.stuckTransactions {
+				if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice()) <= 0 {
+					block.AddTxWithChain(b.blockchain, tx)
+				} else {
+					remainingTx = append(remainingTx, tx)
+				}
+			}
```

Comment:
nit: 
```
                        for _, tx := range append(b.pendingBlock.Transactions(), b.stuckTransactions...) {
				if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice()) <= 0 {
					block.AddTxWithChain(b.blockchain, tx)
				} else {
					remainingTx = append(remainingTx, tx)
				}
			}
```
Can this work with a single loop? Not saving anything just less lines of code

Reply:
thanks for the suggestion, its neat!